backend/src/email/services/custom-domain.service.ts

```diff
@@ -0,0 +1,148 @@
+import dns from 'dns'
+import AWS from 'aws-sdk'
+import escapeHTML from 'escape-html'
+import config from '@core/config'
+
+import { EmailFromAddress } from '@email/models'
+import { MailService } from '@core/services'
+import logger from '@core/logger'
+import { formatFromAddress } from '@core/utils/from-address'
+
+const [, region] = config.get('mailOptions.host').split('.')
```

Comment:
Thought about this while testing the frontend. We might have an issue deriving the region this way because we're using different SES credentials on the backend and the workers. The domain might be verified on the worker's SES region but not on the backend's SES region.